core/common/src/test/java/alluxio/resource/DynamicResourcePoolTest.java

```diff
@@ -0,0 +1,248 @@
+    List<Resource> resourceList = new ArrayList<>();
+    for (int i = 0; i < 3; i++) {
+      Resource resource = pool.acquire();
+      resourceList.add(resource);
+      Assert.assertEquals(i, resource.mInteger.intValue());
+    }
+
+    for (Resource resource : resourceList) {
+      pool.release(resource);
+    }
+
+    Set<Integer> resources = new HashSet<>();
+    for (int i = 0; i < 3; i++) {
+      Resource resource = pool.acquire();
+      resources.add(resource.mInteger);
+    }
+    // Make sure we are not creating new resources.
+    for (int i = 0; i < 3; i++) {
+      Assert.assertTrue(resources.contains(i));
+    }
+  }
+
+  /**
+   * Acquire without capacity.
+   *
+   * @throws Exception if anything fails
+   */
+  @Test
+  public void acquireWithoutCapacity() throws Exception {
+    TestPool pool = new TestPool(DynamicResourcePool.Options.defaultOptions().setMaxCapacity(1));
+    List<Resource> resourceList = new ArrayList<>();
+    boolean timeout = false;
+    try {
+      Resource resource = pool.acquire();
+      resourceList.add(resource);
+      Assert.assertEquals(0, resource.mInteger.intValue());
+
+      resource = pool.acquire(1, TimeUnit.SECONDS);
+      resourceList.add(resource);
+    } catch (TimeoutException e) {
+      timeout = true;
+    }
+    Assert.assertEquals(1, resourceList.size());
+    Assert.assertTrue(timeout);
+  }
+
+  /**
+   * Tests the logic that invalid resource won't be acquired.
+   *
+   * @throws Exception if anything fails
+   */
+  @Test
+  public void UnHealhyResource() throws Exception {
+    TestPool pool = new TestPool(DynamicResourcePool.Options.defaultOptions());
+    Resource resource = pool.acquire();
+    Assert.assertEquals(0, resource.mInteger.intValue());
+    resource.setInteger(Resource.INVALID_RESOURCE);
+
+    pool.release(resource);
+    resource = pool.acquire();
+
+    // The 0-th resource is not acquired because it is unhealthy.
+    Assert.assertEquals(1, resource.mInteger.intValue());
+  }
+
+  /**
+   * Tests the logic that the recently used resource is preferred.
+   *
+   * @throws Exception if anything fails
+   */
+  @Test
+  public void acquireRentlyUsed() throws Exception {
+    ManualClock manualClock = new ManualClock();
+    TestPool pool = new TestPool(DynamicResourcePool.Options.defaultOptions(), manualClock);
+    List<Resource> resourceList = new ArrayList<>();
+    resourceList.add(pool.acquire());
+    resourceList.add(pool.acquire());
+    resourceList.add(pool.acquire());
+
+    pool.release(resourceList.get(2));
+    pool.release(resourceList.get(0));
+
+    // Sleep 1.5 seconds.
+    manualClock.addTimeMs(1500);
+    pool.release(resourceList.get(1));
+
+    for (int i = 0; i < 10; i++) {
+      Resource resource = pool.acquire();
+      Assert.assertEquals(1, resource.mInteger.intValue());
+      pool.release(resource);
+    }
+  }
+
+  /**
+   * @throws Exception if anything fails
```

Comment:
You don't need `@throws` for tests javadoc. Basically if the test name is explainable, no javadoc is needed at all.